src/jdk.jshell/share/classes/jdk/jshell/execution/JdiExecutionControl.java

```diff
@@ -90,7 +90,7 @@ public void redefine(ClassBytecodes[] cbcs)
             vm().redefineClasses(rmp);
         } catch (EngineTerminationException ex) {
             throw ex;
-        } catch (Exception ex) {
+        } catch (Exception | LinkageError ex) {
```

Comment:
How was ClassFormatError previously thrown here? Was it wrapped in some exception?

Reply:
It was simply thrown out of the redefine method, as the stacktrace in the summary suggests. Note that the javadoc for `redefineClasses` is very clear it may throw `ClassFormatError`:
https://docs.oracle.com/en/java/javase/25/docs/api/jdk.jdi/com/sun/jdi/VirtualMachine.html#redefineClasses(java.util.Map)